migration/v0.46/prepare/40-preserve-current-os-setup.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+HERE="$(dirname "$(readlink -f "${0}")")"
+ROOT="$(readlink -f "${HERE}/../../../")"
+
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+if [[ "${CK8S_CLUSTER}" =~ ^(sc|both)$ ]]; then
+  log_info "Moving current Opensearch configuration to override config"
+  yq_add sc .opensearch "$(yq4 ".opensearch" <(yq_merge "${CK8S_CONFIG_PATH}/defaults/sc-config.yaml" "${CK8S_CONFIG_PATH}/sc-config.yaml" "${CK8S_CONFIG_PATH}/common-config.yaml" | yq4 -o json))"
```

Comment:
Yes!
```suggestion
  yq_add sc .opensearch "$(yq_merge "${CK8S_CONFIG_PATH}/defaults/sc-config.yaml" "${CK8S_CONFIG_PATH}/sc-config.yaml" "${CK8S_CONFIG_PATH}/common-config.yaml" | yq4 -o json .opensearch)"
```
As a small simplification.